packages/core/getUserAvailability.ts

```diff
@@ -187,27 +187,67 @@ export async function getUserAvailability(
     source: query.withSource ? a.source : undefined,
   }));
 
-  const bookings = busyTimes.filter((busyTime) => busyTime.source?.startsWith(`eventType-${eventType?.id}`));
+  let bookings = busyTimes.filter((busyTime) => busyTime.source?.startsWith(`eventType-${eventType?.id}`));
 
   const bookingLimits = parseBookingLimit(eventType?.bookingLimits);
+  const durationLimits = parseDurationLimit(eventType?.durationLimits);
+
+  // PER_YEAR has been taken care individually inside getBusyTimesFromBookingLimits and getBusyTimesFromDurationLimits
+  const intervalLimitKeys: (keyof IntervalLimit)[] = ["PER_MONTH", "PER_WEEK", "PER_DAY"];
+  let startDate: Dayjs = dateFrom;
+  let endDate: Dayjs = dateTo;
+
+  // should only get these details if limits are in place and we ahve eventTypeId
+  if (eventTypeId && (bookingLimits || durationLimits)) {
+    for (const key of intervalLimitKeys) {
+      if ((bookingLimits && key in bookingLimits) || (durationLimits && key in durationLimits)) {
+        const filter = key.split("_")[1].toLowerCase() as "week" | "month" | "day";
+        startDate = dayjs.min(startDate, startDate.startOf(filter));
+        endDate = dayjs.max(endDate, endDate.endOf(filter));
+      }
+    }
+
+    const currentEventBookingsBeforeDateFromPromise = getEventBookingsForPeriod({
+      startTime: startDate.toISOString(),
```

Comment:
getBusy function fetches the data for `dateFrom - dateTo`

the following code fetches for `startDate - dateFrom` and `dateTo - endDate`